<launch>

  <arg name="app_filepath"/>
  <arg name="ros_frame_id"/>
  <arg name="run_test" default="true"/>

  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>

  <node name="application_tf_handler" pkg="evd_ros_core" type="application_tf_handler.py" output="screen">
    <param name="ros_frame_id" value="$(arg ros_frame_id)"/>
  </node>

  <node name="robot_control_server" pkg="evd_ros_core" type="robot_control_server.py" output="screen"/>

  <node name="trace_processor" pkg="evd_ros_core" type="trace_processor.py" output="screen"/>

  <node name="waypoint_processor" pkg="evd_ros_core" type="waypoint_processor.py" output="screen"/>

  <node name="program_verifier" pkg="evd_ros_core" type="program_verifier.py" output="screen"/>

  <node name="grasp_verifier" pkg="evd_ros_core" type="grasp_verifier.py" output="screen">
    <param name="app_filepath" value="$(arg app_filepath)"/>
  </node>

  <!-- For Debug -->
  <include if="$(arg run_test)" file="$(find evd_ros_core)/launch/test_core_application.launch">
    <arg name="ros_frame_id" value="world"/>
  </include>

</launch>
